### DISCUSSION TITLE
Raspberry Pi OS - Bullseye support

### DISCUSSION THREAD

**Discussion by ghost** [dsmrreader/dsmr-reader#1513] [Questions & answers / How-to's] [answered]
### Description

Aangezien vandaag [Raspberry Pi OS - Bullseye](https://www.raspberrypi.com/news/raspberry-pi-os-debian-bullseye/) is uitgekomen.
Kon ik het als pseudo Tweakers niet laten om Raspberry Pi OS - Bullseye eens te testen met dsmr-reader.
dsmr-reader lijkt allemaal naar behoren te werken op Bullseye. 

Ondernomen stappen:
- Manuele, volledige back-up gemaakt van de dsmr databank.
- Verse installatie Raspberry Pi OS - Bullseye op het micro-sd kaartje gezet.
- De dsmr  'Installatie: Stapsgewijs' uitgevoerd inclusief 'Optioneel: Database back-up terugzetten'.

Ik heb nog een volledige systeemback-up mocht e.e.a. achteraf toch niet helemaal lekker werken.

### DSMR-reader version

4.19

**Comment by dennissiemensma** [comment]
Bedankt voor je probeersel en fijn dat het goed lijkt te werken! Ik zou zeggen kijk het even een weekje aan of je toch niets vreemds ziet. In principe zou een OS-upgrade hooguit issues kunnen geven bij het installeren van packages.

Welke Python-versie komt nu trouwens standaard mee met Bullseye? Al zou het niet uit moeten maken gezien t/m Python 3.9 en zelfs 3.10 goed zou moeten werken met DSMR-reader.

**Reply by ghost** [reply]
@dennissiemensma Bullseye zit op Python 3.9.2 en PostgreSQL 13 (Debian naam 13+225)
Heb even in de logs gekeken. Er staat een melding over de onderstaande python packages.

Kijkend naar de schrijf datum, zijn beide packages meegekomen met de distributie release op het imagebestand van Raspberry Pi OS - Bullseye d.d. 2021-10-30. 

DSMR op Bullseye draait inmiddels een gemiddelde werkdag in uren. 
Tot nu toe loopt alles soepel, zonder problemen. Behoorlijk wat electriciteit verbruikt aangezien vrouwlief diverse hartige gerechten heeft bereid in een electrische oven. Als ik het verbruik vergelijk met vergelijkbare meet momenten dan is de registratie hiervan accuraat. (Al twijfelde ik hier al niet aan). Het valt meer onder de noemer. Kijk het even een weekje aan.

>  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.26.5
    Not uninstalling urllib3 at /usr/lib/python3/dist-packages, outside environment /home/dsmr/.virtualenvs/dsmrreader
    Can't uninstall 'urllib3'. No files were found to uninstall.
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
    Not uninstalling requests at /usr/lib/python3/dist-packages, outside environment /home/dsmr/.virtualenvs/dsmrreader
    Can't uninstall 'requests'. No files were found to uninstall.

**Comment by dennissiemensma** [comment]
Oke helder! En die `urllib3` melding geeft aan dat die de versie van het OS pakt. Opzich prima en in DSMR-reader v5.x gaat dat toch weer op de kop.

**Reply by ghost** [reply]
Raspberry Pi OS - Bullseye draait ongeveer 9 dagen na dato nog steeds naar behoren.
@dennissiemensma Mocht je nog distro specifieke dingen willen weten dan houd ik mij aanbevolen.

**Comment by dennissiemensma** [comment, marked as answer]
Bedankt voor je update! Ik denk dat je niet specifieks hoeft te doen. Over het algemeen komt elk OS weer met nieuwe versies van programma's die zelf weer hun eigen verbeteringen hebben. Ik zou er zelf zo min mogelijk handmatig aan veranderen, tenzij iets overduidelijk niet werkt.
